libimage/filters.go

```diff
@@ -47,18 +49,14 @@ func (i *Image) applyFilters(ctx context.Context, filters map[string][]filterFun
 
 // filterImages returns a slice of images which are passing all specified
 // filters.
-func (r *Runtime) filterImages(ctx context.Context, images []*Image, options *ListImagesOptions) ([]*Image, error) {
+func (r *Runtime) filterImages(ctx context.Context, images []*Image, options *ListImagesOptions, filters filtersMap, tree *layerTree) ([]*Image, error) {
 	if len(options.Filters) == 0 || len(images) == 0 {
```

Comment:
This works, but the `len(options.Filters)` here kind of breaks the abstraction now — `options` has already been interpreted by the compilation, authoritative data for _this_ operation comes in the compiled `filters`.

I think the condition can just be removed, AFAICS `compileImageFilters` correctly returns a no-op filter. Or maybe it can be moved inside `compileImageFilters`. Either way I think the `options` parameter here would be better removed.

(The `len(images) == 0` check here is now unnecessary, the loop does the same thing anyway.)